### ISSUE TITLE
[FEATURE] Leasing in Ride

### ISSUE THREAD

**Issue by CryptoBadBoy** [wavesplatform/Waves#2575]
**Abstract**
Leasing is important function in the waves. But RIDE don't support leasing transaction and does not allow to receive  leasing balance by address. Need functions that will give opportunity for use leasing in RIDE.

**Motivation and Purposes**
In now not allow create LeaseTransaction and CancelLeaseTransaction in ScriptResult. I suggest add LeaseTransaction, CancelLeaseTransaction and leaseBalance function. This help create new finance logic for dapp.

**Specification**

1) leaseBalance(address: Address) - function which return leasing balance by address.

Example invokeScript function that send token to user with positive leasing balance:
```
@Callable(i)
function withdrawToken(){
    let balance = leaseBalance(i.call)
    TransferSet([ScriptTransfer(i.call, balance, RewordTokenId)])
}
```

2) LeaseTransaction(recipient: Address, amount: Amount) - send waves to leasing.
CancelLeaseTransaction(txId: ByteVector) - cancel leasing by txId.

Example where key, dataValue, address, value, id are variables:
```
ScriptResult(
    WriteSet([
        DataEntry(key, dataValue)
    ]),
    LeasingSet([
        LeaseTransaction(address, value),
        CancelLeaseTransaction(id)
    ])
)
```

Need to expand ScriptResult and make more options:
1) ScriptResult(WriteSet, TransferSet) 
2) ScriptResult(WriteSet, LeasingSet)
3) ScriptResult(TransferSet, LeasingSet)
4) ScriptResult(WriteSet, TransferSet, LeasingSet)

**Backwards Compatibility**
Can your proposition affect any existing features?
Not, this does not affect old functions.

**Comment by pivoo81** [comment]
Yes, we plan to implement such functionality.

**Comment by DrBlast** [comment]
Implemented in version 1.3.x
